.github/workflows/test.yml

```diff
@@ -0,0 +1,28 @@
+name: Test
+
+on: [pull_request]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [10.x, 12.x]
+
+    steps:
+    - uses: actions/cache@v2
+      with:
+        path: ~/.npm
+        key: ${{ runner.os }}-${{ matrix.node-version }}-node-${{ hashFiles('**/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.node-version }}-node-     
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
```

Comment:
This should probably be run only if there was a cache miss, otherwise the cache will be pointless (`npm ci` deletes `node_modules` before running)

Reply:
Oh wait, I just realised we cache `.npm` rather than `node_modules` so this above is bad advice, we can't skip `npm ci`